client/layout/guided-tours/config-elements.js

```diff
@@ -0,0 +1,288 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import classNames from 'classnames';
+import { translate } from 'i18n-calypso';
+import {
+	find,
+	mapValues,
+	omit,
+	property,
+} from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import Button from 'components/button';
+import ExternalLink from 'components/external-link';
+import {
+	posToCss,
+	getStepPosition,
+	getValidatedArrowPosition,
+	targetForSlug
+} from './positioning';
+
+const contextTypes = Object.freeze( {
+	next: PropTypes.func.isRequired,
+	quit: PropTypes.func.isRequired,
+	isValid: PropTypes.func.isRequired,
+	tour: PropTypes.string.isRequired,
+	tourVersion: PropTypes.string.isRequired,
+	step: PropTypes.string.isRequired,
+	lastAction: PropTypes.object,
+} );
+
+export class Tour extends Component {
+	static propTypes = {
+		name: PropTypes.string.isRequired
+	}
+
+	static childContextTypes = contextTypes;
+
+	getChildContext() {
+		const { next, quit, isValid, lastAction, tour, tourVersion, step } = this.tourMeta;
+		return { next, quit, isValid, lastAction, tour, tourVersion, step };
+	}
+
+	constructor( props, context ) {
+		super( props, context );
+		this.setTourMeta( props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		this.setTourMeta( nextProps );
+	}
+
+	setTourMeta( props ) {
+		const { next, quit, isValid, lastAction, name, version, stepName } = props;
+		this.tourMeta = { next, quit, isValid, lastAction, tour: name, tourVersion: version, step: stepName };
+	}
+
+	render() {
+		const { children, context, isValid, lastAction, stepName } = this.props;
+		const nextStep = find( children, stepComponent =>
+			stepComponent.props.name === stepName );
+		const isLastStep = nextStep === children[ children.length - 1 ];
+
+		if ( ! nextStep || ! isValid( context ) ) {
+			return null;
+		}
+
+		return React.cloneElement( nextStep, { isLastStep, isValid, lastAction } );
```

Comment:
`isValid` and `lastAction` are already carried in `context`, so they shouldn't be passed here. Likewise, `Step`'s methods should look for those properties in `context` rather than `props`.